core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2190,6 +2190,13 @@ wasm_set_exception(WASMModuleInstance *module_inst, const char *exception)
     }
 
 #if WASM_ENABLE_THREAD_MGR != 0
+
+#if WASM_ENABLE_SHARED_MEMORY
+    if (exception) {
+        notify_stale_threads_on_exception(
+            (WASMModuleInstanceCommon *)module_inst);
+    }
+#endif
```

Comment:
Should move these lines to be after L2204: spread exception firstly to let other threads get exception, and then notify them, so for other thread, if it is atomic waiting and is notified, and if there is another atomic waiting operation, in `wasm_runtime_atomic_wait`, it can detect the exception and return -1 directly. Or it might atomic wait again.

Reply:
makes sense, done